### PR TITLE
maint, breaking: drop support for jupyterlab 2

### PR DESCRIPTION
**Pull request by consideRatio**
This PR is based on top of #393 as I'm modifying the docs also modified there. This PR includes only one commit, [369e5b](https://github.com/jupyterhub/jupyter-server-proxy/pull/394/commits/b369e5bb54105372ded580e84afddc2b7a4e831b).

With this PR, the JupyterLab extension will still be published as an NPM package to @jupyterhub/jupyter-server-proxy, but we will not test against it or ensure future functioning, or document how to install it etc. Dropping support for jupyterlab 2 was proposed to go with the 4.0.0 release planned in(#388 as jupyterlab 4 is coming out and we are to support both jupyterlab 3 and 4 already.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ Server via the container.
 
 ### Requirements
 
-- `jupyterlab>=2` or `notebook`
+Either `jupyterlab>=3` or `notebook<7` is required.
```
Comment (manics) [inline README.md:57]: Shouldn't notebook>=7 also work, since it's based on JupyterLab?

Comment (consideRatio) [reply README.md:57]: Maybe hmm maybe, I just saw a lot of tests against notebook<7 specifically as compared to any version of notebook.

Should i adjust this to be as it was, without the <7 part?

**Comment by bollwyvl** [comment]
Can start testing with betas, but otherwise who knows?

**Review by manics — APPROVED** [review]

**Comment by consideRatio** [comment]
Going for a merge and a changelog PR to get moving on the release efforts!

**Pull request merged**